src/wamr/wamr.cc

```diff
@@ -108,6 +108,7 @@ class Wamr : public WasmVm {
 
   WasmStorePtr store_;
   WasmModulePtr module_;
+  WasmSharedModulePtr shared_module_;
```

Comment:
Dumb question (I'm new to the WAMR implementation): why does `shared_module_` need to be a field of `Wamr`? I.e., why can't the call to `wasm_module_share()` that produces the `WasmSharedModulePtr` happen on-demand in `Wamr::clone()`?

Reply:
That follows the existing convention from other Wasm engines, where we serialize in main thread during load, and then deserialize it when starting worker threads.

Like you've said, it doesn't have to be done that way, but if we want to change that, then it should be done in a subsequent PR for all supported Wasm engines.

Reply:
`shared_module_`  is a some kind of seed in this context. Either `wasm_module_obtain(store, wasm_module_share(module))`  or creating a shareable seed and "cloning" from it. I thought the second one is more natural than the first. 

The first one is also a good choice if treating `module_` as a seed. But in that case, we won't need `wasm_shared_module_t` and should `wasm_module_obtain(store, module)` directly.

Reply:
Thanks for the explanations! Current code SGTM.